### PR TITLE
[Testnet] Airdrop Fixes

### PR DESCRIPTION
**Pull request by mrtpain**
* Fixes bug where `AmountToTransfer` on the `Call` method was incorrectly set to the `AmountToDistribute` property when 0crs should be sent.
* Makes CanRegsiter a publicly callable method
* Updates to README

**hash**
```
470da40fca440583f7b22d6048dd37f53ad8cfd403376c876422a7f921239c04
```

**Bytecode**
```
4D5A90000300000004000000FFFF0000B800000000000000400000000000000000000000000000000000000000000000000000000000000000000000800000000E1FBA0E00B409CD21B8014CCD21546869732070726F6772616D2063616E6E6F742062652072756E20696E20444F53206D6F64652E0D0D0A2400000000000000504500004C010200544241B40000000000000000E00022200B013000001000000002000000000000362F0000002000000040000000000010002000000002000004000000000000000400000000000000006000000002000000000000030040850000100000100000000010000010000000000000100000000000000000000000E42E00004F000000000000000000000000000000000000000000000000000000004000000C000000C82E00001C0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000200000080000000000000000000000082000004800000000000000000000002E746578740000003C0F0000002000000010000000020000000000000000000000000000200000602E72656C6F6300000C000000004000000002000000120000000000000000000000000000400000420000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000182F0000000000004800000002000500D4230000F40A00000100000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000FE0203280500000A0205166AFE037201000070280600000A0205280500000602042803000006020E0428070000060202280700000A6F0800000A28090000062A4602280900000A721F0000706F0A00000A2A4A02280900000A721F000070036F0B00000A2A4602280900000A72490000706F0C00000A2A4A02280900000A7249000070036F0D00000A2A4602280900000A72610000706F0C00000A2A4A02280900000A7261000070036F0D00000A2A4602280900000A72730000706F0A00000A2A4A02280900000A7273000070036F0B00000A2A4602280900000A727F0000706F0C00000A2A4A02280900000A727F000070036F0D00000A2A4602280900000A72A70000706F0C00000A2A4A02280900000A72A7000070036F0D00000A2A7202280900000A72CD000070038C07000001280E00000A6F0F00000A2A7602280900000A72CD000070038C07000001280E00000A046F1000000A2A7A0228060000062C1402281100000A6F1200000A022806000006FE052A172A4A0202280700000A6F0800000A28160000062A8A02280700000A6F0800000A022808000006281300000A2C08020328160000062A162A0000133002003D0000000100001102280900000A72A70000706F0C00000A0A062D270228100000062D1F02280A000006166A361502280400000602280A0000065C0A0206280D000006062AB602280700000A6F0800000A022808000006281400000A2C02162A0202281100000A6F1200000A2807000006172A0013300600CC000000020000110202280700000A6F0800000A280E00000672E3000070281500000A2D080228100000062C02162A0228130000060A198D0800000125160228080000068C07000001A2251702280700000A6F0800000A8C07000001A22518068C10000001A20B02022802000006166A72F500007007166A281600000A0C02086F1700000A7201000070280600000A0202280700000A6F0800000A720F010070280F000006021203FE1503000002120302280700000A6F0800000A7D030000041203720F0100707D0400000409280100002B172A13300300760000000300001103022808000006281300000A2C02162A0203280E000006281900000A2C160228100000062C0E02280A0000060228040000063702162A0202280A000006176AD7280B000006020372E3000070280F000006021200FE15030000021200037D03000004120072E30000707D0400000406280100002B172A000042534A4201000100000000000C00000076342E302E33303331390000000005006C00000010040000237E00007C0400001404000023537472696E677300000000900800002001000023555300B0090000100000002347554944000000C00900003401000023426C6F620000000000000002000001571DA201090A000000FA01330016000001000000100000000300000004000000160000000F0000001900000002000000040000000300000001000000060000000C000000010000000200000001000000010000000000CA0101000000000006000B014C0206003A014C020600F70039020F006C0200000A007F03EC020A006700EC020A00DC02EC0206008D03DC010A009403EC0206005D00DC010A002B01EC020A004800EC020A008E00EC0206008001DC010A009101EC0206000E00DC01000000001500000000000100010001001000F50100001500010001000A0110008701000029000300170051804003C60051803303C6000600A703C90006006A03C60050200000000086183302CD0001009020000000008608870221000500A220000000008108A002D7000500B520000000008608CD0359000600C720000000008108DD03DD000600DA20000000008608A20159000700EC20000000008108AF01DD000700FF2000000000860813022100080011210000000081081D02D7000800242100000000860803035900090036210000000081081B03DD0009004921000000008108B30059000A005B21000000008108CA00DD000A006E210000000086004F03E2000B008B210000000081006003E8000C00A921000000008600270285000E00C8210000000086002A0285000E00DB21000000008600A403EF000E000022000000008600E10059000F004922000000008600E30185000F007822000000008600C40385000F0050230000000081005801EF000F00000001007B0000000200B90200000300ED0300000400BC01000001006D01000001006D01000001006D01000001006D01000001006D01000001006D0100000100E40200000100E40200000200710300000100B20300000100B203090033020100110033020600190033020A005900330206002900330210002900BD031600290051001C0061000802210029009F0026006900CE022B006900D902310069000100380069000B003D0071007803430069007301490069007D014E002900980154007900FD0159003900F9035D00390005045D007100050473002900C501790049007B02850029008D018900710035009A000E000400A8000E000800B9002E000B00FE002E00130007012E001B002601610023002F01650069009500020001000000A402F5000000E103FA000000B301FA0000002102F50000001F03FA000000E400FA0002000200030001000300030002000400050001000500050002000600070001000700070002000800090001000900090002000A000B0001000B000B0002000C000D0001000D000D000480000000000000000000000000000000007F0300000400000000000000000000009F001E00000000000100020001000000000000000000EC02000000000300020031009000000000000047657455496E7436340053657455496E743634003C4D6F64756C653E0053797374656D2E507269766174652E436F72654C69620049734E756C6C4F725768697465537061636500494D657373616765006765745F4D6573736167650056616C7565547970650049536D617274436F6E7472616374537461746500736D617274436F6E74726163745374617465004950657273697374656E745374617465006765745F50657273697374656E745374617465006765745F416D6F756E74546F44697374726962757465007365745F416D6F756E74546F4469737472696275746500476574416D6F756E74546F446973747269627574650044656275676761626C6541747472696275746500436F6D70696C6174696F6E52656C61786174696F6E7341747472696275746500496E6465784174747269627574650052756E74696D65436F6D7061746962696C6974794174747269627574650041646452656769737472616E74457865637574650076616C756500476574537472696E6700536574537472696E67005374617475734C6F670049426C6F636B006765745F426C6F636B006765745F456E64426C6F636B007365745F456E64426C6F636B00656E64426C6F636B0043616C6C00536D617274436F6E74726163742E646C6C0053797374656D00436C6F7365526567697374726174696F6E0041697264726F70006765745F4E756D626572006765745F53656E646572006765745F4F776E6572007365745F4F776E65720043616E5265676973746572002E63746F720053797374656D2E446961676E6F73746963730053797374656D2E52756E74696D652E436F6D70696C6572536572766963657300446562756767696E674D6F646573006765745F53756363657373006765745F546F6B656E436F6E747261637441646472657373007365745F546F6B656E436F6E74726163744164647265737300746F6B656E436F6E74726163744164647265737300476574416464726573730053657441646472657373006164647265737300537472617469732E536D617274436F6E747261637473006765745F4E756D6265724F6652656769737472616E7473007365745F4E756D6265724F6652656769737472616E74730046756E64656453746174757300456E726F6C6C6564537461747573004765744163636F756E74537461747573005365744163636F756E745374617475730073746174757300466F726D617400536D617274436F6E7472616374004F626A65637400495472616E73666572526573756C740041646452656769737472616E740072656769737472616E7400417373657274005769746864726177006765745F546F74616C537570706C79007365745F546F74616C537570706C7900746F74616C537570706C79006F705F457175616C697479006F705F496E657175616C6974790000001D41007300730065007200740020006600610069006C00650064002E00002954006F006B0065006E0043006F006E00740072006100630074004100640064007200650073007300001754006F00740061006C0053007500700070006C007900001145006E00640042006C006F0063006B00000B4F0077006E006500720000274E0075006D006200650072004F006600520065006700690073007400720061006E0074007300002541006D006F0075006E00740054006F00440069007300740072006900620075007400650000155300740061007400750073003A007B0030007D00001145004E0052004F004C004C004500440000195400720061006E007300660065007200460072006F006D00000D460055004E0044004500440000000000EF027E56C33E9640BFAF8D15FB91235C0004200101080320000105200101111105200101121905200201020E0420001231042000111D0420001235052001111D0E062002010E111D0420010B0E052002010E0B0500020E0E1C0420010E0E052002010E0E042000123D0320000B07000202111D111D0307010B0907040B1D1C1225110C050002020E0E0B20051225111D0B0E1D1C0B0320000206300101011E00040A01110C040701110C040001020E087CEC85D7BEA7798E1045004E0052004F004C004C00450044000C460055004E0044004500440002060E0306111D092004011219111D0B0B05200101111D042001010B0520010E111D06200201111D0E05200102111D042800111D0328000B0801000800000000001E01000100540216577261704E6F6E457863657074696F6E5468726F7773010801000200000000000401000000000000000000000000000000100000000000000000000000000000000C2F00000000000000000000262F0000002000000000000000000000000000000000000000000000182F0000000000000000000000005F436F72446C6C4D61696E006D73636F7265652E646C6C0000000000FF250020001000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000002000000C000000383F00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000
```

### CODE REVIEW

**Comment by mrtpain** [comment]
@codingupastorm Found a bug which led to some other small improvements....new hash and bytecode provided.

Sorry about that, when you have time, review/merge. Your help getting this to testnet would be appreciated as well.

Manually tested most everything, just got hung up on the withdraw failures. Hoping I caught the issue.

**Comment by codingupastorm** [comment]
Don't be sorry @mrtpain , we missed it too!

Will have a chat to the right people. 

I need to get notifications set up for this repo...

File: Testnet/Airdrop/Airdrop/Airdrop.cs [hunk 1]
```diff
@@ -134,14 +134,14 @@ public bool Withdraw()
...
-        if (invalidAccountStatus || CanRegister)
+        if (invalidAccountStatus || CanRegister())
         {
             return false;
         }
 
         var amountToDistribute = GetAmountToDistribute();
         var transferParams = new object[] { Owner, Message.Sender, amountToDistribute };
-        var result = Call(TokenContractAddress, amountToDistribute, "TransferFrom", transferParams);
+        var result = Call(TokenContractAddress, 0, "TransferFrom", transferParams);
```
Comment (codingupastorm) [inline Testnet/Airdrop/Airdrop/Airdrop.cs]: Nice. I definitely should have picked this up last time.

**Comment by XelsAdmin** [comment]
Thanks for your efforts guys. We were busy with some other pieces of stuff and got no time to review your pull request. We'll soon review the code. Thanks again for your contribution.

**Pull request closed without merging**